nemo/collections/common/tokenizers/huggingface/auto_tokenizer.py

```diff
@@ -0,0 +1,213 @@
+            text_ += text[i]
+        i += 1
+    return text_
+
+
+def remove_spaces(text):
+    text = text.replace("( ", "(")
+    text = text.replace(" )", ")")
+    text = text.replace("[ ", "[")
+    text = text.replace(" ]", "]")
+    text = text.replace(" / ", "/")
+    text = text.replace("„ ", "„")
+    text = text.replace(" - ", "-")
+    text = text.replace(" ' ", "'")
+    text = re.sub(r'([0-9])( )([\.,])', '\\1\\3', text)
+    text = re.sub(r'([\.,])( )([0-9])', '\\1\\3', text)
+    text = re.sub(r'([0-9])(:)( )([0-9])', '\\1\\2\\4', text)
+    text = text.replace(" %", "%")
+    text = text.replace("$ ", "$")
+    text = re.sub(r'([^0-9])(,)([0-9])', '\\1\\2 \\3', text)
+    return text
+
+
+class AutoTokenizer(TokenizerSpec):
+    '''
+        Wrapper of HuggingFace AutoTokenizer https://huggingface.co/transformers/model_doc/auto.html#autotokenizer.
+    '''
+
+    def __init__(
+        self,
+        pretrained_model_name: str,
+        vocab_file: Optional[str] = None,
+        do_lower_case: Optional[bool] = None,
+        mask_token: Optional[str] = None,
+        bos_token: Optional[str] = None,
+        eos_token: Optional[str] = None,
+        pad_token: Optional[str] = None,
+        sep_token: Optional[str] = None,
+        cls_token: Optional[str] = None,
+        unk_token: Optional[str] = None,
+    ):
+
+        """
+        Args:
+            pretrained_model_name: corresponds to HuggingFace-AutoTokenizer's 'pretrained_model_name_or_path' input argument. 
+                For more details please refer to https://huggingface.co/transformers/_modules/transformers/tokenization_auto.html#AutoTokenizer.from_pretrained. 
+                The list of all supported models can be found here: ALL_PRETRAINED_CONFIG_ARCHIVE_MAP
+            vocab_file: path to file with vocabulary which consists
+                of characters separated by '\n'.
+            do_lower_case: determines if data should be lower cased before tokenization.
+            mask_token: mask token 
+            bos_token: the beginning of sequence token
+            eos_token: the end of sequence token. Usually equal to sep_token
+            pad_token: token to use for padding
+            sep_token: token used for separating sequences
+            cls_token: class token. Usually equal to bos_token
+            unk_token: token to use for unknown tokens
+        """
+
+        if pretrained_model_name not in ALL_PRETRAINED_CONFIG_ARCHIVE_MAP:
+            raise ValueError(f"{pretrained_model_name} not a huggingface pretrained model")
+
+        if vocab_file is not None:
+            self.tokenizer = AUTOTOKENIZER.from_pretrained(
+                pretrained_model_name_or_path=pretrained_model_name, vocab_file=vocab_file, do_lower_case=do_lower_case
+            )
+        else:
+            self.tokenizer = AUTOTOKENIZER.from_pretrained(
+                pretrained_model_name_or_path=pretrained_model_name, do_lower_case=do_lower_case
+            )
```

Comment:
We used do_lower_case for SPE only, in the old setup and it was ignored if it didn't match the pretrained_model_name. Do we need it here?

Reply:
yes as discussed in meeting

Reply:
@titu1994 it depends some huggingface tokenizers do not have do_lower_case member. So yes i will remove do_lower_case entirely from AutoTokenizer, as you suggested